processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -166,6 +171,26 @@ public Filter toFilter()
   }
 
   @Override
+  public RangeSet<String> getDimensionRangeSet(String dimension)
+  {
+    if (!Objects.equals(getDimension(), dimension) || getExtractionFn() != null || alphaNumeric) {
+      return null;
+    }
+    RangeSet<String> retSet = TreeRangeSet.create();
+    Range<String> range;
+    if (getLower() == null) {
+      range = isUpperStrict() ? Range.lessThan(getUpper()) : Range.atMost(getUpper());
```

Comment:
Does "Range" deal with nulls the same way filters do (`null == ""`?) If not then there needs to be `Strings.nullToEmpty` at various places here. Please also add some tests for the null case if they do not already exist.